lib/controller.rb

```diff
@@ -59,6 +59,18 @@ class SlowFood < Sinatra::Base
     erb :menu
   end
 
+  enable :sessions
+
+  post '/menu' do
+    @dish = Dish.get(params[:dishid].to_i)
+    @dish.save
+    @cart = Cart.new
+    @cart = @dish
```

Comment:
In this line, you are setting `@cart` equal to `@dish` - that is, you are replacing the cart object with the dish object. That is not what you want.